### PR TITLE
Display previews of all supported media types

### PR DESCRIPTION
**Pull request by oparoz**
**This is beta code and still contains a lot of calls to console.log OC_Log::write**

**This has been rewritten as an AppFramework app, but the same logic applies, so voice your opinion here**

This modification does the following:
- Introduces a list of media types that a Gallery apps should be able to show

```
    'image/png'
    'image/jpeg'
    'image/gif'
    'image/x-xbitmap'
    'image/bmp'
    'image/svg+xml'
    'image/tiff'
    'image/x-dcraw'
    'image/svg+xml'
    'application/x-photoshop'
    'application/illustrator'
    'application/postscript'
```
- Adds a few media types to the slideshow, so that a user can quickly see what the file is about

```
    'application/font-sfnt'
    'application/x-font'
```
- Checks the list of potential media types against the list of media types supported by the system before populating any images array
- Displays GIFs without any conversion so that animations are not lost
- Displays SVGs natively if the SVG provider is not enabled
- Tries to fail gracefully while we wait for https://github.com/owncloud/core/pull/12546
- Tries to harmonize code across all the files in the ajax folder
- Removes some (apparently) dead code (thumbnail.php)
- Maintains compatibility with OC7, assuming [it has been patched to support more bitmap previews](https://gist.github.com/oparoz/229c07bbcb8a209481f7)

Known issues
- The Preview or thumbnail generation methods have not been touched by this PR (yet), so thumbnails of small documents may appear distorted, just like when using master
- If the SVG provider is not turned on, the Gallery will try to display SVGs natively, without checking if it's supported by the browser
- The slideshow will still hang if it encounters a file which has failed to be converted and the fallback doesn't work in this case either. This can be fixed via this core PR: https://github.com/owncloud/core/pull/12546

### CODE REVIEW

**Comment by oparoz** [comment]
@libasys - If you want to give it a go ;)

**Comment by oparoz** [comment]
@georgehrke - This is previews related ;)

File: ajax/getimages.php [hunk 1]
```diff
@@ -1,71 +1,124 @@
...
-$path = null;
 
-if (isset($_GET['token'])) {
-	$token = $_GET['token'];
+// $slideshow will determine if we return images for the Gallery or media types for the Javascript slideshow
+$slideshow = isset($_GET['slideshow']) ? $_GET['slideshow'] : false;
+$token = isset($_GET['token']) ? $token = $_GET['token']: false;
+
+// This builds a list of supported media types
+function getSupportedMimes($slideshow) {
```
Comment (libasys) [inline ajax/getimages.php:17]: It make more sense to write a class than single functions!

**Comment by oparoz** [comment]
Closing this as app has been entirely re-written. The current branch is dead as it's using too many private APIs, contains duplication, dead code and has a couple of security issues.

File: ajax/getimages.php (hunk 1 above)
Comment (oparoz) [inline ajax/getimages.php:17]: The goal was to add a few missing features, not to fix all its problems. Having functions made it easier to see what was going on.

**Pull request closed without merging**